src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -567,6 +571,12 @@ internal void CancelConnectAsync()
                 }
                 else
                 {
+                    if (SocketsTelemetry.IsEnabled(EventLevel.Warning))
+                    {
+                        SocketsTelemetry.Log.ConnectCancelled();
+                        SocketsTelemetry.Log.ConnectStop();
```

Comment:
Seems like this should be on the same plan as ConnectFailed, with just one log at the call site.